### PR TITLE
Add metrics support  #SMA-800

### PR DESCRIPTION
**Pull request by AnastasiiaVashchuk**

### CODE REVIEW

File: metrics/metrics-config.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export const NumberOfQueuedTransfers = new prometheus.Gauge({
+    name: "numberOfQueuedTransfers",
+    help: "number of transfers that correspond to the number of requests for faucet's funds"
+})
+
+export const TransferAttempts = new prometheus.Counter({
+    name: "transferAttempts",
+    help: "attempts to execute transfers from the queue",
+    labelNames: ["status"]
+});
```
Comment (RomanBrodetski) [inline metrics/metrics-config.ts:26]: let's keep the metric names case consistent. I think normally snake_case is used

File: app/app.ts [hunk 2]
```diff
@@ -83,20 +91,27 @@ async function processTweets(): Promise<void> {
...
-      const response = await getTweets(query);
+
+      const response = await getTweets(query, "mentions?since_id");
       if (response["meta"]["result_count"] !== 0) {
         sinceId = response["meta"]["newest_id"];
         consoleLog(`sinceId is updated to ${sinceId}`);
 
         for (let tweet of response["data"]) {
           consoleLog(`Faucet has received the tweet with id ${tweet["id"]}`);
+          TweetCounter.labels({ tweet_category: "total mentions" }).inc();
```
Comment (RomanBrodetski) [inline app/app.ts:102]: we don't need a separate counter for total - we can compute it with Prometheus if needed (`faucet request` + `other mention tweets`)

File: app/app.ts [hunk 3]
```diff
@@ -83,20 +91,27 @@ async function processTweets(): Promise<void> {
...
 
           if (await validateTweet(tweet["text"], tweet["id"])) {
-            consoleLog(`Tweet with id ${tweet["id"]} is valid for requesting funds`);
+            consoleLog(
+              `Tweet with id ${tweet["id"]} is valid for requesting funds`
+            );
+            // valid tweet means tweet that corresponds tweet's text pattern
+            TweetCounter.labels({ tweet_category: "faucet request" }).inc();
```
Comment (RomanBrodetski) [inline app/app.ts:109]: it's better not to have whitespace in label names. Let's just have `faucet`/`other`

**Review by RomanBrodetski — COMMENTED** [review]
Overall looks really good!

File: metrics/metrics-config.ts (hunk 1 above)
Comment (AnastasiiaVashchuk) [reply metrics/metrics-config.ts:26]: done

File: app/app.ts (hunk 2 above)
Comment (AnastasiiaVashchuk) [reply app/app.ts]: fixed

File: app/app.ts [hunk 4]
```diff
@@ -304,17 +342,21 @@ async function startSendingMoney(queueNumber: number): Promise<void> {
...
   );
-  consoleLog(`${provider.connection.url} provider instance is initialized in startUpdatingAvailableQueues`);
+  consoleLog(
+    `${provider.connection.url} provider instance is initialized in startUpdatingAvailableQueues`
+  );
 
   while (true) {
     try {
+      queueOfTransfers = sendMoneyQueue.reduce((preSum, transfers) => transfers.length + preSum, 0)
+      console.log("queueOfTransfers", queueOfTransfers.toString())
```
[the remark this answers is not in the document]
Comment (AnastasiiaVashchuk) [reply app/app.ts]: fixed

File: app/app.ts [hunk 5]
```diff
@@ -62,29 +72,30 @@ async function setSinceId() {
...
   await setSinceId();
+  let response;
   while (true) {
     try {
       const query = `users/${ZKSYNC_ID}/mentions?since_id=${sinceId}`;
       consoleLog(`Request for @zksync's mentions`);
-      
-      const response = await getTweets(query);
+
+      response = await getTweets(query, "mentions?since_id");
```
Comment (RomanBrodetski) [inline app/app.ts:98]: I think `"mentions?since_id"` is not a good label name. Let's not use any special characters there. Why not just "mentions"?

**Review by RomanBrodetski — APPROVED** [review]

**Pull request merged**